lib/possible_bipartition.rb

```diff
@@ -1,4 +1,62 @@
+#an undirected graph is bipartite if there exists partition into Left & Right;
+#it must be 2 colorable 
+#it has no odd length cycles  
+
+#assign floofi to source dog 
+#color all neighbors with spotti 
+#color all neighbors neighbors with floofi
+#continue this way until all dogs are colored 
+# while assigning colors, if we find a neighbor which is colored with the same color as the current vertex, then the graph is not bipartite 
 
 def possible_bipartition(dislikes)
-  raise NotImplementedError, "possible_bipartition isn't implemented yet"
-end
+  return true if dislikes.length <= 2 
+  #we use a queue to keep track of all the neighbors once we visit a node
+  queue = []
+  dog_types = []
+  total_dogs = dislikes.length 
+
+  #we can start by putting the first node in the queue
+  #it doesnt matter which node we use to start 
+  #we can just put in 0 to our queue, which indicates the index of our first start dog 
+  queue << 0
+
+  # So what if one of the dislikes arrays are empty? 
+  # That means the nodes are isolated & can go anywhere   
+  total_dogs.times do |index|
+    if !dog_types[index] 
+      queue.push(index)
+      dog_types[index] = "floofi"
+    end 
+  end 
```

Comment:
I don't think you should be setting the `dog_types` here to start with.  

Instead I'd put a loop which goes through all the pets and if they are not already marked in `dog_types` add them to the queue and mark them as floofi.  Then put your `while !queue.empty?` inside that loop.

This way you don't pre-mark a dog floofi until you've made sure no other node connects to it directly or indirectly.